### ISSUE TITLE
saving state of comparison page to url

### ISSUE THREAD

**Issue by jjgao** [cBioPortal/cbioportal#6165]
comparison/?sessionId=xxxxx&unselectedGroups=group1,group2&deletedGroups=group3,group4&addedGroups=group5&groupOrder=group5,group6

**Comment by jjgao** [comment]
Ok to do delay it until after 3.0.0.

**Comment by jjgao** [comment]
based on discussion in https://github.com/cBioPortal/cbioportal/issues/5525#issuecomment-489733778

**Comment by adamabeshouse** [comment]
ohhh i see

would the groupOrder parameter replace our strategy of saving it into the session?

**Comment by jjgao** [comment]
@adamabeshouse I think we can keep them in the url and at the same time can also save them into session.

**Comment by adamabeshouse** [comment]
@jjgao I think what you're saying is that when the order is unsaved, we keep it in the URL. when they save it to the session, we remove the order parameter. so then we should also adjust the warning to be less urgent

**Comment by jjgao** [comment]
@adamabeshouse sounds good!

**Comment by adamabeshouse** [comment]
addedGroup would have to contain a JSON stringified entire group data. I think it would be clunky. I'll implement the others

**Comment by jjgao** [comment]
@adamabeshouse I though the added group is saved back to the study? If so, would it work tot add the group ID to the url?

**Comment by adamabeshouse** [comment]
@jjgao yes that would be okay in the case that the added group is saved back to the study. but if the user isnt logged in that wouldn't be an option. in my opinion its a bit confusing but it's doable

**Comment by jjgao** [comment]
@adamabeshouse I see. Maybe we don't have to support `addedGroups` when a user is not logged in. I think the "create group" functionality is not so useful anyway if a user is not logged in - they can only do comparisons with non-overlapping groups of samples from pie chart or bar chart.

**Comment by adamabeshouse** [comment]
In discussion with @jjgao and @alisman we decided that it was confusing to have two sources of truth - session and URL. So we decided that

(1) Adding and deleting a group will instantly save to a new session. Wasted sessions not a big deal since this doesn't happen that much anyway.
(2) Drag group order and selection will be stored only in the URL, never in the session.
(3) So the "unsaved changes"->Save button header will no longer appear.
